projects/core/src/cms/facade/page-meta.service.ts

```diff
@@ -60,23 +86,62 @@ export class PageMetaService {
    * @param metaResolver
    */
   protected resolve(metaResolver: PageMetaResolver): Observable<PageMeta> {
-    const resolveMethods: Observable<PageMeta>[] = Object.keys(
-      this.resolverMethods
-    )
-      .filter((key) => metaResolver[this.resolverMethods[key]])
-      .map((key) =>
-        metaResolver[this.resolverMethods[key]]().pipe(
+    const resolverMethods = this.getResolverMethods();
+    const resolvedData: Observable<PageMeta>[] = Object.keys(resolverMethods)
+      .filter((key) => metaResolver[resolverMethods[key]])
+      .map((key) => {
+        return metaResolver[resolverMethods[key]]().pipe(
           map((data) => ({
             [key]: data,
           }))
-        )
-      );
+        );
+      });
 
-    return uniteLatest(resolveMethods).pipe(
+    return uniteLatest(resolvedData).pipe(
       map((data) => Object.assign({}, ...data))
     );
   }
 
+  /**
+   * Returns an object with resolvers. The object properties represent the `PageMeta` property, i.e.:
+   *
+   * ```
+   * {
+   *   title: 'resolveTitle',
+   *   robots: 'resolveRobots'
+   * }
+   * ```
+   *
+   * This list of resolvers is filtered for CSR vs SSR processing since not all resolvers are
+   * relevant during browsing.
+   */
+  protected getResolverMethods(): { [property: string]: string } {
+    let resolverMethods = {};
+    const configured = this.pageMetaConfig?.pageResolvers?.resolvers;
+    if (configured) {
+      configured
+        // filter the resolvers to avoid unnecessary processing in CSR
+        .filter((resolver) => {
+          return (
+            !isPlatformBrowser(this.platformId) ||
+            !(
+              resolver.disabledInCsr &&
+              !(
+                isDevMode() &&
+                this.pageMetaConfig?.pageResolvers?.enableInDevMode
+              )
+            )
+          );
```

Comment:
Can we simplify this condition? I hope this is an equivalent (could you double check please, if it was your intention)?:
```ts
!isPlatformBrowser(this.platformId) || // show when it's not CSR OR
!resolver.disabledInCsr ||             // show when it's not disabled in CSR OR
(                                      // show anyway in dev mode, when `enableInDevMode` is true 
  isDevMode() &&
  this.pageMetaConfig?.pageResolvers?.enableInDevMode
)
```

Reply:
much nicer, thanks!